doc/examples/type_hints.rst

```diff
@@ -0,0 +1,237 @@
+
+.. _type_hints-example:
+
+Type Hints
+===========
+
+As of version 4.1, PyMongo ships with `type hints`_.
+
+With type hints, Python type checkers can easily find bugs before they reveal themselves in your code.  If your IDE is configured to use type hints,
```

Comment:
Maybe put a link here for how to configure type hints in popular IDEs?

Reply:
Done